controlplane/pkg/remote/endpoint_selector_service.go

```diff
@@ -143,15 +138,24 @@ func (cce *endpointSelectorService) validateConnection(ctx context.Context, conn
 		return err
 	}
 
-	result, err := cce.pluginRegistry.GetConnectionPluginManager().ValidateConnection(ctx, conn)
-	if err != nil {
+	ipCtx := conn.GetContext().GetIpContext()
+	if err := cce.validateIPAddress(ipCtx.GetSrcIpAddr(), "srcIP"); err != nil {
 		return err
 	}
 
-	if result.GetStatus() != pluginapi.ConnectionValidationStatus_SUCCESS {
-		return errors.Errorf(result.GetErrorMessage())
-	}
+	return cce.validateIPAddress(ipCtx.GetDstIpAddr(), "dstIP")
+}
 
+func (cce *endpointSelectorService) validateIPAddress(ip, ipName string) error {
```

Comment:
`validateIPAddress` needs `ipName` just to construct error message. I think we can build error on the upper level and avoid passing `ipName` here

Reply:
In that case we would have to duplicate the code generating an error, and also to distinguish the intersection from other error (returned from cce.prefixes.Intersect(ip))